loopy/codegen/__init__.py

```diff
@@ -152,7 +152,10 @@ def __init__(self, iname, length, space):
 
 
 class SeenFunction(ImmutableRecord):
-    """
+    """This is used to track functions that emerge late during code generation,
+    e.g. C functions to realize arithmetic. No connection with
+    :class:`~loopy.kernel.function_interface.InKernelCallable`.
```

Comment:
This is the main change here. @kaushikcfd, do you think we can get away with declaring these two features independent?

Reply:
Do we end up having kernel callables for these calls?

Reply:
Things like `pow` which are provided by C99 are resolved as KernelCallables, but not modulo, divs. I guess we could completely strip off SeenFunction and make callables for the modulo and divs as well.

Reply:
I think I'm leaning towards leaving them as is right now. If they're just used to realize built-in things, I kind of don't see a pressing need for these functions to become full-on callables. They could legitimately be viewed as a target property, just like `+` in `a + b`. If we elevate them to become callables, then that would kind of have to happen in a target-specific pass pre-codegen, which sounds expensive. What do you think?

Reply:
> I kind of don't see a pressing need for these functions to become full-on callables. 

Agreed, at least the space to which loopy targets to i.e. providing a transformable IR, for that purpose realizing primitive operations as callables so deep into the pipeline doesn't help much.

Reply:
OK then, in it goes. IMO it's also not a super risky decision. Since it's internal, we can probably change our minds.

Reply:
Thanks for your thoughts here!